src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const res = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = Object.assign({}, currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        currentState = Object.assign({}, currentState);
```

Comment:
Similarly, there is no need to create a new object with Object.assign here since you are not adding any new properties. You can directly delete the keys from currentState.